ext/snmp/snmp.c

```diff
@@ -1612,6 +1619,8 @@ PHP_METHOD(SNMP, close)
 	php_snmp_object *snmp_object;
 	zval *object = ZEND_THIS;
 
+	// Possibly using fetching object to avoid closing pointlessly,
+	// albeit netsnmp_session_free already checks the session (master?) ?
 	snmp_object = Z_SNMP_P(object);
```

Comment:
I would leave that as is for now. `netsnmp_session_free()` (which should better be named `snmp_session_free()` since it's not a symbol of Net-SNMP) is called several times, and checking for an empty session is likely there.

Could still clean up in the master branch later.